test/integration/scheduler/queue_test.go

```diff
@@ -178,89 +180,154 @@ func TestSchedulingGates(t *testing.T) {
 // TestCoreResourceEnqueue verify Pods failed by in-tree default plugins can be
 // moved properly upon their registered events.
 func TestCoreResourceEnqueue(t *testing.T) {
```

Comment:
This table-driven test is copied from https://github.com/kubernetes/kubernetes/pull/122292.
This test has to be added to prove this bug existed and is fixed by this PR.